stacker/blueprints/base.py

```diff
@@ -179,6 +200,11 @@ def resolve_variable(var_name, var_def, provided_variable, blueprint_name):
     var_type = var_def.get("type")
     value = validate_variable_type(var_name, var_type, value)
 
+    allowed_values = var_def.get('allowed_values')
+    valid = validate_allowed_values(allowed_values, value)
+    if not valid:
+        raise UnallowedValue(var_name, value, allowed_values)
```

Comment:
Do you think we need a special exception here?  Seems like a `ValueError`, though I guess at the point where we move to all stacker exceptions being from a base exception this might be useful.